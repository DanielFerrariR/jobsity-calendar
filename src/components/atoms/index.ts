export { default as Box } from './box'
export { default as Button } from './button'
export { default as Paper } from './paper'
export { default as Typography } from './typography'
export { default as TextField } from './text_field'
export { default as Container } from './container'
export { default as AppBar } from './appbar'
export { default as IconButton } from './icon_button'
export { default as Toolbar } from './toolbar'
export { default as Image } from './image'
export { default as Grid } from './grid'
export { default as Link } from './link'
export { default as Checkbox } from './checkbox'
export { default as Accordion } from './accordion'
export { default as AccordionSummary } from './accordion_summary'
export { default as AccordionDetails } from './accordion_details'
export { default as Tooltip } from './tooltip'
export { default as Modal } from './modal'
export { default as MenuItem } from './menu_item'
export { default as ListItemIcon } from './list_item_icon'
export { default as ListItemText } from './list_item_text'
export { default as ListItemSecondaryAction } from './list_item_secondary_action'
export { default as MenuList } from './menu_list'
export { default as Popover } from './popover'
export { default as Fade } from './fade'
export { default as Backdrop } from './backdrop'
export { default as CircularProgress } from './circular_progress'
export { default as LinearProgress } from './linear_progress'
export { default as Skeleton } from './skeleton'
export { default as List } from './list'
export { default as ListItem } from './list_item'
export { default as ClickAwayListener } from './click_away_listener'
export { default as Badge } from './badge'
export { default as Portal } from './portal'
export { default as Snackbar } from './snackbar'
export { default as Step } from './step'
export { default as StepContent } from './stepContent'
export { default as StepLabel } from './stepLabel'
export { default as Stepper } from './stepper'
export { default as TableBody } from './table_body'
export { default as TableContainer } from './table_container'
export { default as TableHead } from './table_head'
export { default as TableRow } from './table_row'
export { default as Table } from './table'
export { default as TableCell } from './table_cell'
export { default as DateTimePicker } from './date_time_picker'
