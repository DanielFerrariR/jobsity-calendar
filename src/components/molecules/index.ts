export { CustomTableHead, CustomTableBody } from './custom_table'
export { default as Modal } from './modal'
export { default as ReminderCard } from './reminder_card'
