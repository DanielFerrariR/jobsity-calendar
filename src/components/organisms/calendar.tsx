import React, { useState } from 'react'
import {
  Tooltip,
  Paper,
  Typography,
  Button,
  IconButton,
  Box,
  TableContainer,
  Table,
  TableCell,
  TableRow
} from 'src/components/atoms'
import {
  CustomTableHead,
  CustomTableBody,
  ReminderCard
} from 'src/components/molecules'
import { ArrowForwardIos } from '@material-ui/icons'
import { useTheme, useMediaQuery } from '@material-ui/core'
import ListRemindersModal from './list_reminders_modal'
import CreateReminderModal from './create_reminder_modal'

const Calendar: React.FC = () => {
  const [openCreateReminderModal, setOpenCreateReminderModal] = useState(false)
  const [openListRemindersModal, setOpenListRemindersModal] = useState<
    false | Date
  >(false)
  const theme = useTheme()
  const xsAndDown = useMediaQuery(theme.breakpoints.down('xs'))
  const today =
    process.env.NODE_ENV === 'test' || window.Cypress
      ? new Date('2020/10/20')
      : new Date()
  const [year, setYear] = useState(today.getFullYear())
  const [month, setMonth] = useState(today.getMonth())
  const monthNames = [
    'January',
    'February',
    'March',
    'April',
    'May',
    'June',
    'July',
    'August',
    'September',
    'October',
    'November',
    'December'
  ]
  const weekDaysNames = [
    'Sunday',
    'Monday',
    'Tuesday',
    'Wednesday',
    'Thursday',
    'Friday',
    'Saturday'
  ]

  const setToday = () => {
    setMonth(today.getMonth())
    setYear(today.getFullYear())
  }

  const setPreviousMonth = () => {
    if (month > 0) {
      setMonth(month - 1)
    } else {
      setYear(year - 1)
      setMonth(11)
    }
  }

  const setNextMonth = () => {
    if (month < 11) {
      setMonth(month + 1)
    } else {
      setYear(year + 1)
      setMonth(0)
    }
  }

  const createBodyLine = (current: number) => {
    const firstDay = new Date(year, month).getDay()
    const previousMonth = month === 0 ? 11 : month - 1
    const daysInPastMonth = 32 - new Date(year, previousMonth, 32).getDate()
    const daysInMonth = 32 - new Date(year, month, 32).getDate()
    const bodyLine = []

    for (let count = current * 7; count < current * 7 + 7; count += 1) {
      if (count < firstDay) {
        const newDay = daysInPastMonth + count - firstDay + 1
        const newMonth = month > 0 ? month - 1 : 11
        const newYear = month > 0 ? year : year - 1
        const newDate = new Date(newYear, newMonth, newDay)
        bodyLine.push(
          <TableCell
            key={count}
            onClick={() => setOpenListRemindersModal(newDate)}
            data-testid={`calendar-table-cell-past-${newDay}`}
          >
            <Typography fontWeight="500" color={theme.palette.grey[500]} mb={1}>
              {newDay}
            </Typography>
            <ReminderCard
              date={newDate}
              testId={`reminder-card-past-${newDay}`}
            />
          </TableCell>
        )
      } else if (count - firstDay + 1 > daysInMonth) {
        const newDay = count - daysInMonth - firstDay + 1
        const newMonth = month < 11 ? month + 1 : 0
        const newYear = month < 11 ? year : year + 1
        const newDate = new Date(newYear, newMonth, newDay)

        bodyLine.push(
          <TableCell
            key={count}
            onClick={() => setOpenListRemindersModal(newDate)}
            data-testid={`calendar-table-cell-future-${newDay}`}
          >
            <Typography fontWeight="500" color={theme.palette.grey[500]} mb={1}>
              {newDay}
            </Typography>
            <ReminderCard
              date={newDate}
              testId={`reminder-card-future-${newDay}`}
            />
          </TableCell>
        )
      } else {
        const newDay = count - firstDay + 1
        const isWeekend = count === current * 7 || count === current * 7 + 6
        const isToday =
          count - firstDay + 1 === today.getDate() &&
          month === today.getMonth() &&
          year === today.getFullYear()
        let dayColor = ''
        const newDate = new Date(year, month, newDay)

        if (isToday) {
          dayColor = theme.palette.error.main
        } else if (isWeekend) {
          dayColor = theme.palette.primary.main
        }

        bodyLine.push(
          <TableCell
            key={count}
            onClick={() => setOpenListRemindersModal(newDate)}
            data-testid={`calendar-table-cell-current-${newDay}`}
          >
            <Typography fontWeight="500" color={dayColor} mb={1}>
              {newDay}
            </Typography>
            <ReminderCard
              date={newDate}
              testId={`reminder-card-current-${newDay}`}
            />
          </TableCell>
        )
      }
    }

    return bodyLine
  }

  const createBody = () => {
    const body = []

    for (let count = 0; count < 6; count += 1) {
      body.push(<TableRow key={count}>{createBodyLine(count)}</TableRow>)
    }

    return body
  }

  return (
    <>
      <Paper p={2}>
        <Box
          display="flex"
          alignItems="center"
          mb={4}
          flexDirection={xsAndDown && 'column'}
          position="relative"
        >
          <Box
            display="flex"
            alignItems="center"
            position={!xsAndDown && 'absolute'}
            mb={xsAndDown && 2}
            width={1}
            justifyContent="space-between"
            flexDirection={xsAndDown && 'column'}
          >
            <Box mb={xsAndDown && 2}>
              <Tooltip title="Back">
                <IconButton
                  aria-label="Back"
                  onClick={setPreviousMonth}
                  disabled={month === 0 && year === 1970}
                  data-testid="calendar-button-back"
                >
                  <ArrowForwardIos style={{ transform: 'rotate(180deg)' }} />
                </IconButton>
              </Tooltip>
              <Tooltip title="Forward">
                <IconButton
                  aria-label="Forward"
                  onClick={setNextMonth}
                  mr={1}
                  disabled={month === 11 && year === 9999}
                  data-testid="calendar-button-forward"
                >
                  <ArrowForwardIos />
                </IconButton>
              </Tooltip>
              <Button
                onClick={setToday}
                data-testid="calendar-button-set-today"
              >
                Today
              </Button>
            </Box>
            <Button
              ml={1}
              onClick={() => setOpenCreateReminderModal(true)}
              data-testid="calendar-button-create-reminder"
            >
              Create reminder
            </Button>
          </Box>
          <Typography
            variant="h6"
            textAlign="center"
            width={1}
            ml={1}
            data-testid="calendar-typography-date"
          >
            {monthNames[month]} {year}
          </Typography>
        </Box>
        <TableContainer>
          <Table style={{ minWidth: 1200 }}>
            <CustomTableHead>
              <TableRow>
                {weekDaysNames.map((each, index) => {
                  return (
                    <TableCell key={index}>
                      <Typography textAlign="center" fontWeight="500">
                        {each}
                      </Typography>
                    </TableCell>
                  )
                })}
              </TableRow>
            </CustomTableHead>
            <CustomTableBody>{createBody()}</CustomTableBody>
          </Table>
        </TableContainer>
      </Paper>
      <ListRemindersModal
        open={openListRemindersModal}
        setOpen={setOpenListRemindersModal}
      />
      <CreateReminderModal
        open={openCreateReminderModal}
        setOpen={setOpenCreateReminderModal}
      />
    </>
  )
}

export default Calendar
