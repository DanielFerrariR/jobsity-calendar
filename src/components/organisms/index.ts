export { default as Calendar } from './calendar'
export { default as ListRemindersModal } from './list_reminders_modal'
export { default as CreateReminderModal } from './create_reminder_modal'
export { default as EditReminderModal } from './edit_reminder_modal'
