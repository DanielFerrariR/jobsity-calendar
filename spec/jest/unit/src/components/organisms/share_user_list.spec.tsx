import React from 'react'
import { ShareUserList } from 'src/components/organisms'
import { render } from 'spec/jest/config/test_utils'
import { axe } from 'jest-axe'
import { userData } from 'spec/jest/fixtures'

describe('testing ShareUserList component', () => {
  const shareUserListForm = {
    search: ''
  }
  const shareUserListErrors = {
    search: false
  }
  const sharedUsers = [
    {
      id: 1,
      username: 'Daniel',
      admin: true
    }
  ]

  test('should snapshot works with english language', () => {
    const { container } = render(
      <ShareUserList
        shareUserListForm={shareUserListForm}
        setShareUserListForm={(): void => {}}
        shareUserListErrors={shareUserListErrors}
        setShareUserListErrors={(): void => {}}
        sharedUsers={sharedUsers}
        setSharedUsers={(): void => {}}
      />,
      {
        initialState: { user: userData }
      }
    )

    expect(container).toMatchSnapshot()
  })

  test('should snapshot works with portuguese language', () => {
    const { container } = render(
      <ShareUserList
        shareUserListForm={shareUserListForm}
        setShareUserListForm={(): void => {}}
        shareUserListErrors={shareUserListErrors}
        setShareUserListErrors={(): void => {}}
        sharedUsers={sharedUsers}
        setSharedUsers={(): void => {}}
      />,
      {
        language: 'pt',
        initialState: { user: userData }
      }
    )

    expect(container).toMatchSnapshot()
  })

  test('should snapshot works with spanish language', () => {
    const { container } = render(
      <ShareUserList
        shareUserListForm={shareUserListForm}
        setShareUserListForm={(): void => {}}
        shareUserListErrors={shareUserListErrors}
        setShareUserListErrors={(): void => {}}
        sharedUsers={sharedUsers}
        setSharedUsers={(): void => {}}
      />,
      {
        language: 'es',
        initialState: { user: userData }
      }
    )

    expect(container).toMatchSnapshot()
  })

  test('should the page be accessible', async () => {
    const { container } = render(
      <ShareUserList
        shareUserListForm={shareUserListForm}
        setShareUserListForm={(): void => {}}
        shareUserListErrors={shareUserListErrors}
        setShareUserListErrors={(): void => {}}
        sharedUsers={sharedUsers}
        setSharedUsers={(): void => {}}
      />,
      {
        initialState: { user: userData }
      }
    )
    const results = await axe(container)

    expect(results).toHaveNoViolations()
  })
})
