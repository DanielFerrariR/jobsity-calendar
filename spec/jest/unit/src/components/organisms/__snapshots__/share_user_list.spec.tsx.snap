// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing ShareUserList component should snapshot works with english language 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root Component-root Component-root"
  >
    <form
      class="MuiBox-root MuiBox-root Component-root Component-root"
    >
      <div
        class="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root WithStyles(ForwardRef(TextField))-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-required MuiInputLabel-required"
          data-shrink="true"
          data-testid="textfield-share-user-list-label"
          for="textfield-share-user"
          id="textfield-share-user-label"
        >
          Username/E-mail
          <span
            aria-hidden="true"
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
        >
          <input
            aria-invalid="false"
            autocomplete="off"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
            data-testid="textfield-share-user-list"
            id="textfield-share-user"
            maxlength="100"
            placeholder="Type the user name or e-mail"
            required=""
            type="text"
            value=""
          />
          <button
            aria-label="Search"
            class="MuiButtonBase-root MuiIconButton-root Component-root Component-root Styled(Component)-root Styled(Component)-root WithStyles(ForwardRef(Tooltip))-root WithStyles(ForwardRef(Tooltip))-root"
            data-testid="iconbutton-share-user-list-search"
            tabindex="0"
            title="Search"
            type="submit"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled PrivateNotchedOutline-legendNotched"
            >
              <span>
                Username/E-mail
                 *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </form>
    <ul
      class="MuiList-root Component-root Component-root MuiList-padding"
    >
      <li
        class="MuiListItem-container"
      >
        <div
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root Component-root MuiListItem-gutters MuiListItem-button MuiListItem-secondaryAction"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemText-root WithStyles(ForwardRef(ListItemText))-root WithStyles(ForwardRef(ListItemText))-root"
            data-testid="listitemtext-user-list-0"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              <div
                class="MuiBox-root MuiBox-root Component-root Component-root"
              >
                Daniel
              </div>
            </span>
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
        <div
          class="MuiListItemSecondaryAction-root WithStyles(ForwardRef(ListItemSecondaryAction))-root"
        >
          <div
            class="MuiBox-root MuiBox-root Component-root Component-root"
          >
            <div
              class="MuiBox-root MuiBox-root Component-root Component-root"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorSecondary WithStyles(ForwardRef(Checkbox))-root WithStyles(ForwardRef(Checkbox))-root Styled(WithStyles(ForwardRef(Checkbox)))-root Styled(WithStyles(ForwardRef(Checkbox)))-root PrivateSwitchBase-checked MuiCheckbox-checked MuiIconButton-colorSecondary"
                data-testid="checkbox-admin-0"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    checked=""
                    class="PrivateSwitchBase-input"
                    data-indeterminate="false"
                    id="checkbox-admin-1"
                    type="checkbox"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
              <label
                class="MuiTypography-root Component-root Component-root MuiTypography-body1"
                for="checkbox-admin-1"
              >
                Administrator
              </label>
            </div>
            <button
              aria-label="Delete"
              class="MuiButtonBase-root MuiIconButton-root Component-root Component-root Styled(Component)-root Styled(Component)-root WithStyles(ForwardRef(Tooltip))-root WithStyles(ForwardRef(Tooltip))-root"
              data-testid="iconbutton-share-user-list-delete-0"
              tabindex="0"
              title="Delete"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;

exports[`testing ShareUserList component should snapshot works with portuguese language 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root Component-root Component-root"
  >
    <form
      class="MuiBox-root MuiBox-root Component-root Component-root"
    >
      <div
        class="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root WithStyles(ForwardRef(TextField))-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-required MuiInputLabel-required"
          data-shrink="true"
          data-testid="textfield-share-user-list-label"
          for="textfield-share-user"
          id="textfield-share-user-label"
        >
          Nome de usuário/E-mail
          <span
            aria-hidden="true"
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
        >
          <input
            aria-invalid="false"
            autocomplete="off"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
            data-testid="textfield-share-user-list"
            id="textfield-share-user"
            maxlength="100"
            placeholder="Digite o nome ou e-mail"
            required=""
            type="text"
            value=""
          />
          <button
            aria-label="Buscar"
            class="MuiButtonBase-root MuiIconButton-root Component-root Component-root Styled(Component)-root Styled(Component)-root WithStyles(ForwardRef(Tooltip))-root WithStyles(ForwardRef(Tooltip))-root"
            data-testid="iconbutton-share-user-list-search"
            tabindex="0"
            title="Buscar"
            type="submit"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled PrivateNotchedOutline-legendNotched"
            >
              <span>
                Nome de usuário/E-mail
                 *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </form>
    <ul
      class="MuiList-root Component-root Component-root MuiList-padding"
    >
      <li
        class="MuiListItem-container"
      >
        <div
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root Component-root MuiListItem-gutters MuiListItem-button MuiListItem-secondaryAction"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemText-root WithStyles(ForwardRef(ListItemText))-root WithStyles(ForwardRef(ListItemText))-root"
            data-testid="listitemtext-user-list-0"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              <div
                class="MuiBox-root MuiBox-root Component-root Component-root"
              >
                Daniel
              </div>
            </span>
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
        <div
          class="MuiListItemSecondaryAction-root WithStyles(ForwardRef(ListItemSecondaryAction))-root"
        >
          <div
            class="MuiBox-root MuiBox-root Component-root Component-root"
          >
            <div
              class="MuiBox-root MuiBox-root Component-root Component-root"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorSecondary WithStyles(ForwardRef(Checkbox))-root WithStyles(ForwardRef(Checkbox))-root Styled(WithStyles(ForwardRef(Checkbox)))-root Styled(WithStyles(ForwardRef(Checkbox)))-root PrivateSwitchBase-checked MuiCheckbox-checked MuiIconButton-colorSecondary"
                data-testid="checkbox-admin-0"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    checked=""
                    class="PrivateSwitchBase-input"
                    data-indeterminate="false"
                    id="checkbox-admin-1"
                    type="checkbox"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
              <label
                class="MuiTypography-root Component-root Component-root MuiTypography-body1"
                for="checkbox-admin-1"
              >
                Administrador
              </label>
            </div>
            <button
              aria-label="Deletar"
              class="MuiButtonBase-root MuiIconButton-root Component-root Component-root Styled(Component)-root Styled(Component)-root WithStyles(ForwardRef(Tooltip))-root WithStyles(ForwardRef(Tooltip))-root"
              data-testid="iconbutton-share-user-list-delete-0"
              tabindex="0"
              title="Deletar"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;

exports[`testing ShareUserList component should snapshot works with spanish language 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root Component-root Component-root"
  >
    <form
      class="MuiBox-root MuiBox-root Component-root Component-root"
    >
      <div
        class="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root WithStyles(ForwardRef(TextField))-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-required MuiInputLabel-required"
          data-shrink="true"
          data-testid="textfield-share-user-list-label"
          for="textfield-share-user"
          id="textfield-share-user-label"
        >
          Nombre de usuario/E-mail
          <span
            aria-hidden="true"
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
        >
          <input
            aria-invalid="false"
            autocomplete="off"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
            data-testid="textfield-share-user-list"
            id="textfield-share-user"
            maxlength="100"
            placeholder="Escriba el nombre o e-mail"
            required=""
            type="text"
            value=""
          />
          <button
            aria-label="Buscar"
            class="MuiButtonBase-root MuiIconButton-root Component-root Component-root Styled(Component)-root Styled(Component)-root WithStyles(ForwardRef(Tooltip))-root WithStyles(ForwardRef(Tooltip))-root"
            data-testid="iconbutton-share-user-list-search"
            tabindex="0"
            title="Buscar"
            type="submit"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled PrivateNotchedOutline-legendNotched"
            >
              <span>
                Nombre de usuario/E-mail
                 *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </form>
    <ul
      class="MuiList-root Component-root Component-root MuiList-padding"
    >
      <li
        class="MuiListItem-container"
      >
        <div
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root Component-root MuiListItem-gutters MuiListItem-button MuiListItem-secondaryAction"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemText-root WithStyles(ForwardRef(ListItemText))-root WithStyles(ForwardRef(ListItemText))-root"
            data-testid="listitemtext-user-list-0"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              <div
                class="MuiBox-root MuiBox-root Component-root Component-root"
              >
                Daniel
              </div>
            </span>
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
        <div
          class="MuiListItemSecondaryAction-root WithStyles(ForwardRef(ListItemSecondaryAction))-root"
        >
          <div
            class="MuiBox-root MuiBox-root Component-root Component-root"
          >
            <div
              class="MuiBox-root MuiBox-root Component-root Component-root"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorSecondary WithStyles(ForwardRef(Checkbox))-root WithStyles(ForwardRef(Checkbox))-root Styled(WithStyles(ForwardRef(Checkbox)))-root Styled(WithStyles(ForwardRef(Checkbox)))-root PrivateSwitchBase-checked MuiCheckbox-checked MuiIconButton-colorSecondary"
                data-testid="checkbox-admin-0"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    checked=""
                    class="PrivateSwitchBase-input"
                    data-indeterminate="false"
                    id="checkbox-admin-1"
                    type="checkbox"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
              <label
                class="MuiTypography-root Component-root Component-root MuiTypography-body1"
                for="checkbox-admin-1"
              >
                Administrador
              </label>
            </div>
            <button
              aria-label="Eliminar"
              class="MuiButtonBase-root MuiIconButton-root Component-root Component-root Styled(Component)-root Styled(Component)-root WithStyles(ForwardRef(Tooltip))-root WithStyles(ForwardRef(Tooltip))-root"
              data-testid="iconbutton-share-user-list-delete-0"
              tabindex="0"
              title="Eliminar"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
